### PR TITLE
TW-492: Forward still ins tack after forward

### PR DESCRIPTION
**Pull request by nqhhdev**
- Issue: #492 

https://github.com/linagora/twake-on-matrix/assets/99852347/9b6b60e6-7ad4-42d2-9de4-384608d5682d

- Resolved: 

### Web

https://github.com/linagora/twake-on-matrix/assets/99852347/4540b4f2-f08d-4f67-9215-9e7c0f0763fa

### Android

https://github.com/linagora/twake-on-matrix/assets/99852347/c586ea7b-43bc-4c48-b7f0-ce014a11f7cf

### CODE REVIEW

**Comment by sherlockvn** [comment]
You should add before and after demo, so we can understand the issue better

File: lib/pages/chat/chat.dart [hunk 1]
```diff
@@ -1287,3 +1293,8 @@ class ChatController extends State<Chat>
 }
 
 enum EmojiPickerType { reaction, keyboard }
+
+class FowardAgr {
```
Comment (sherlockvn) [inline lib/pages/chat/chat.dart:1297]: What is this ??

File: lib/config/go_routes/go_router.dart [hunk 2]
```diff
@@ -217,11 +218,18 @@ abstract class AppRoutes {
             ),
             GoRoute(
               path: 'forward',
+              redirect: (context, state) {
+                if (state.extra is! ForwardArgument) {
+                  return '${state.fullPath?.replaceAll('forward', '')}';
+                } else {
+                  return '/rooms/forward';
+                }
+              },
```
Comment (sherlockvn) [inline lib/config/go_routes/go_router.dart:227]: Early return:
```suggestion
                if (state.extra is! ForwardArgument) {
                  return '${state.fullPath?.replaceAll('forward', '')}';
                }
               
                 return '/rooms/forward';
                }
              },
```

File: lib/pages/chat/chat.dart (hunk 1 above)
Comment (hoangdat) [reply lib/pages/chat/chat.dart:1297]: we have `copilot` and `autocomplete` please use the full name :))

Comment (nqhhdev) [reply lib/pages/chat/chat.dart:1297]: I'll remove it

**Review by drminh2807 — APPROVED** [review]

**Review by hoangdat — APPROVED** [review]

**Pull request merged**